include/boost/json/value.hpp

```diff
@@ -2832,6 +2832,65 @@ struct key_value_pair
 
 //----------------------------------------------------------
 
+#ifdef BOOST_JSON_DOCS
+
+/** Tuple-like element access.
+    
+    This overload permits the key and value
+    of a `key_value_pair` to be accessed
+    by index. For example:
+
+    @code 
+
+    key_value_pair kvp("num", 42);
+    
+    auto key = get<0>(kvp);
+    auto value = get<1>(kvp);
```

Comment:
We shouldn't use auto in docs, as there is no benefit to doing so and because showing the actual type is more informative for the reader:
```
string_view const key = get<0>(kvp);
value& jv = get<1>(kvp);